docs/configure/sidebar-and-urls.md

```diff
@@ -26,76 +26,100 @@ If you’d prefer to show top-level nodes as folders rather than roots, you can
 
 <CodeSnippets
   paths={[
-    'common/component-story-configuration.js.mdx',
+    'common/other-foo-bar-story.js.mdx',
   ]}
 />
 
 <!-- prettier-ignore-end -->
 
-Given this configuration, the stories file `../src/components/MyComponent.stories.tsx` will get the title `components/MyComponent`.
+Storybook will prioritize the `id` over the title for ID generation if provided and prioritize the `story.name` over the export key for display.
 
-You can further customize the generated title by modifying the stories configuration.
+## CSF 3.0 auto-titles
+
+Storybook 6.4 introduced CSF 3.0 as an experimental feature, allowing you to write stories more compactly. Suppose you're already using this format to write your stories. In that case, you can omit the `title` element from the default export and allow Storybook automatically infer it based on the file's physical location. For example, given the following configuration and story:
 
 <!-- prettier-ignore-start -->
 
 <CodeSnippets
   paths={[
-    'common/component-story-configuration-custom.js.mdx',
+    'common/storybook-main-configuration-src-dir.main-js.js.mdx',
+    'common/component-story-auto-title.csf3-story.js.mdx',
   ]}
 />
 
 <!-- prettier-ignore-end -->
 
-This configuration would match a custom file pattern, and add a custom prefix of Foo to each generated title.
+When Storybook loads, the story can show up in the sidebar as `components/My Component`. While accurate for most cases, it could lead to naming issues when loaded.
 
-## Permalinking to stories
+### Auto-title filename case
 
-By default, Storybook generates an `id` for each story based on the component title and the story name. This `id` in particular is used in the URL for each story, and that URL can serve as a permalink (primarily when you [publish](../sharing/publish-storybook.md) your Storybook).
+Starting with Storybook 6.5, story titles generated automatically no longer rely on Lodash's [startCase](https://lodash.com/docs/#startCase).
+Instead, the file name casing is preserved, allowing additional control over the story title. For example, `components/My Component` will be defined as `components/MyComponent`.
 
-Consider the following story:
+If you need, you can revert to the previous pattern by adding the following configuration:
 
 <!-- prettier-ignore-start -->
 
 <CodeSnippets
   paths={[
-    'common/foo-bar-baz-story.js.mdx',
+    'common/storybook-manager-render-label-stories.js.mdx',
   ]}
 />
 
 <!-- prettier-ignore-end -->
 
-Storybook's ID-generation logic will give this the `id` `foo-bar--baz`, so the link would be `?path=/story/foo-bar--baz`.
+### Auto-title redundant filenames
+
+In addition to improvements to the story file name casing, a new heuristic was introduced, removing the filename if it matches an existing directory name or an `index.stories.js|ts` file. For example, `components/MyComponent/MyComponent.stories.js` is defined as `MyComponent/MyComponent` in the sidebar.
```

Comment:
I had a bit of trouble understanding this paragraph. What do you think of this suggestion?

```suggestion
In addition to improvements to the story file name casing, a new heuristic was introduced, removing redundant names in case the filename has the same name as the directory name, or if it's called `index.stories.js|ts`. For example, before `components/MyComponent/MyComponent.stories.js` was defined as `Components/MyComponent/MyComponent` in the sidebar. Now it will be defined as `Components/MyComponent`.
```